src/kernel/heap2.zig

```diff
@@ -142,14 +193,18 @@ const FreeListAllocator = struct {
     ///     std.mem.Allocator.Error.OutOfMemory - If there wasn't enough free memory to expand into
     ///
     fn resize(allocator: *Allocator, old_mem: []u8, new_size: usize, size_alignment: u29) Allocator.Error!usize {
+        return @fieldParentPtr(FreeListAllocator, "allocator", allocator).resizeInternal(old_mem, new_size, size_alignment);
```

Comment:
How come this has been moved into another function

Reply:
I wanted to test the internal alloc functions explicitly instead of going through the Allocator interface, so I made functions that take a `FreeListAllocator` instead of `Allocator` so that I could do `free_list.alloc(...)`. I've since realised that I could just do `FreeListAllocator.alloc(&free_list.allocator, ...)` instead so I can remove these new functions.